Yank/storage.py

```diff
@@ -0,0 +1,451 @@
+
+    Run some_function() to generate data over an iteration, store each value in a dynamically sized var called "looper"
+    >>> my_storage = StorageInterface('my_store.nc')
+    >>> for i in range(10):
+    >>>     mydata = some_function()
+    >>>     my_storage.looper.append(mydata)
+
+    """
+
+    def __init__(self, file_name, storage_system='netcdf'):
+        """
+        Parameters
+        ----------
+        file_name : string
+            Specify what file name to read and write to on the disk.
+        storage_system : string
+            Specify what type of storage system module to load for interfacing with the file_name
+        """
+
+    """
+    USER FUNCTIONS
+    Methods the user calls on SIDV variables to actually store and read data from disk. These cannot be names of child
+    SIDV instances and will raise an error if you try to use them as such.
+    """
+
+    def write(self, data, protected_write=True):
+        """
+        Write data to a variable which cannot be appended to, nor overwritten (unless specified). This method should
+        be called when you do not want the value stored to change, typically written only once. The protected_write
+        variable will let you overwrite this behavior if you want to use something on disk that is a toggleable flag,
+        but does not change in shape.
+
+        Metadata is added to this variable if possible to indicate this is a write protectable variable.
+
+        This method raises an error if this instance of SIDV is DIRECTORY
+
+        Parameters
+        ----------
+        data
+            This is what will be written to disk, the data will be processed by the STORAGESYSTEM as to how to actually
+            handle the data once the units have been stripped out and assigned to this instance of SIDV.
+        protected_write : boolean, default True
+            Decide whether to check if the variable is already on file if present.
+            If True, no overwrite is allowed and an error is raised if the variable already exists.
+            If False, overwriting is allowed but the variable will still not be able to be appended to.
+
+        """
+
+    def append(self, data):
+        """
+        Write data to a variable who's size changes every time this function is called. The first dimension of the
+        variable on disk will be appended with data after checking the size. If no variable exists, one is created with
+        a dynamic variable size for its first dimension, then the other dimensions are inferred from the data.
+
+        This method raises an error if this instance of SIDV is DIRECTORY
+
+        Parameters
+        ----------
+        data
+            This is what will be written to disk, the data will be processed by the STORAGESYSTEM as to how to actually
+            handle the data once the units have been stripped out and assigned to this instance of SIDV.
+
+        """
+
+    def fetch(self):
```

Comment:
Sorry if I missed it. Does it provide a way to read a particular iteration or the whole array?